app/cdap/components/Metadata/SearchResults/helper.ts

```diff
@@ -0,0 +1,263 @@
+/*
+ * Copyright © 2017 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+import intersection from 'lodash/intersection';
+import { IFilterItem, Operations } from 'components/Metadata/SearchResults/Filters';
+import T from 'i18n-react';
+
+const I18N_PREFIX = 'features.MetadataSearch';
+
+export interface ISearchResult {
+  name: string;
+  type: string;
+  entityTypeState: string;
+  icon: string;
+  datasetExplorable: boolean;
+  queryFound: string[];
+  description: string;
+  createDate: string;
+  datasetType: string;
+}
+
+export interface IParsedResult {
+  results: ISearchResult[];
+  count: number;
+}
+
+interface IProperty {
+  name: string;
+  scope: string;
+  value: string;
+}
+
+export const METADATA_FILTERS = [
+  'Name',
+  'Description',
+  'User tags',
+  'System tags',
+  'User properties',
+  'System properties',
+  'Schema',
+];
+
+export function parseResult(
```

Comment:
Should we add tests for some of these? Also jsdoc them?

Reply:
Done